packages/react-i18n/src/utils/i18n.utils.js

```diff
@@ -47,10 +47,43 @@ export const translate = (lang, i18nNames = {}) => {
 
     const translation = _.get(lang, combineKey, combineKey);
 
-    return sprintf(translation, { ...data, ...i18nNames, number });
+    const translatedResult = sprintf(translation, { ...data, ...i18nNames, number });
+    if (renderers) {
+      return instantiateJSXInsideTranslation(translatedResult, renderers);
+    }
+
+    return translatedResult;
   };
 };
 
+const JSX_REGEX = /(.*?)<(\w+)>(.*)<\/\2+>(.*)/;
+const instantiateJSXInsideTranslation = (translation, renderers) => {
+  const regexResult = JSX_REGEX.exec(translation);
+
+  if (!regexResult) {
+    return [translation];
+  }
+
+  const [_, beforeTagSiblings, component, children, afterTagSiblings] = regexResult;
+  const result = [];
+
+  if (beforeTagSiblings) {
+    result.push(...instantiateJSXInsideTranslation(beforeTagSiblings, renderers));
+  }
+
+  if (children) {
+    const parsedChildren = instantiateJSXInsideTranslation(children, renderers);
+    const renderResult = renderers[component]({ children: parsedChildren });
```

Comment:
Not sure it works with classes 🤔 
```suggestion
    const renderResult = React.createElement(renderers[component], { children: parsedChildren });
```

Reply:
Indeed, you right. I live a bit in the future ! 😄 
Changing that 👍